### PR TITLE
feat(streams): to ArrayBuffer/Blob/Json/Text

### PR DESCRIPTION
**Pull request by lino-levan**
Closes https://github.com/denoland/deno_std/issues/3587.

### CODE REVIEW

File: streams/json_from_readable_stream.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+// This module is browser compatible.
+
+import { textFromReadableStream } from "./text_from_readable_stream.ts";
+
+export function jsonFromReadableStream(
+  readableStream: ReadableStream,
+): Promise<unknown> {
+  return textFromReadableStream(readableStream).then((res) => {
+    console.log(res);
```
Comment (iuioiua) [inline streams/json_from_readable_stream.ts:10]:
```suggestion
```

Comment (lino-levan) [reply streams/json_from_readable_stream.ts:10]: Oh haha, forgot I left that in there. Nice catch.

File: streams/array_buffer_from_readable_stream.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+// This module is browser compatible.
+
+import { concat } from "../bytes/concat.ts";
+
+export async function arrayBufferFromReadableStream(
+  readableStream: ReadableStream,
```
Comment (kt3k) [inline streams/array_buffer_from_readable_stream.ts:7]: Probably better to limit this type to `ReadableStream<Uint8Array>`?

**Comment by kt3k** [comment]
@lino-levan 
In my view `streams/` is going to be mostly about ReadableStreams (especially after deprecating `Reader`/`Writer`-related stuff). So I think `ReadableStream` in API naming might sound redundant. How about `toArrayBuffer`, `toBlob`, `toJson`, and `toText`?

**Review by kt3k — APPROVED** [review]
LGTM!

**Pull request merged**